source/extensions/filters/http/json_to_metadata/filter.cc

```diff
@@ -289,6 +289,15 @@ void Filter::processBody(const Buffer::Instance* body, const Rules& rules,
   }
 
   Json::ObjectSharedPtr body_json = std::move(result.value());
+  // A pure string or number is considered as a valid application/json body, so we should
+  // treat this case as on_missing for all rules but a successful json body.
+  if (!body_json) {
+    ENVOY_LOG(debug, "a valid non-object json body which is error prone");
+    handleAllOnMissing(rules, request_processing_finished_);
+    success.inc();
```

Comment:
Can we update the docs to have this stat specify what it is? 

Also, it's a bit confusing still as it's "success" or rq_success but we're not "succeeding" in the way I'd expect that to mean. Is success then success + transformation success (but invalid non-object?) -- that seems a bit confusing to me.

Reply:
`rq_invalid_json_body` should only indicates the body is not a valid json.
However pure string and number is probably considered as a *valid* json body by consensus. Therefore it should be considered `rq_success` since we succeed to parse the body. yeah it's confused while we comes to this corner case anyway :( 

Maybe it would make more senses if we consider to support empty selectors here to get the pure string/number in the future?
https://github.com/envoyproxy/envoy/blob/81b11f7bf9d94f557b04749a70240c3fcb40fad1/api/envoy/extensions/filters/http/json_to_metadata/v3/json_to_metadata.proto#L81

Reply:
Thanks for updating the doc, it does reflect `rq_success` means json transformation was successful.

> Maybe it would make more senses if we consider to support empty selectors here to get the pure string/number in the future?

I think given my understanding of the filter as it is now, I don't think the semantics of an empty selector would make much sense.